c7n/resources/iam.py

```diff
@@ -1264,7 +1276,11 @@ def process(self, resources):
             self.process_user(client, r)
 
     def process_user(self, client, r):
-        user_options = self.data.get('options', list(self.ORDERED_OPTIONS.keys()))
+        # filter out options that yield the same operation,
+        # see definition of ORDERED_OPTIONS above
+        default_options = {v: k for k, v in self.ORDERED_OPTIONS.items()}.values()
```

Comment:
you're changing this to a regular dictionary and could mess up the order of operations as dictionary keys are hashed when created. Can you change this to a OrderedDict like the original `ORDERED_OPTIONS`

Reply:
No, see the original code below: it goes through all items in `ORDERED_OPTIONS` in original order and skips those who are not in `user_options`. This is, because the user_options from the `policies.yaml` can come in random order.

Reply:
oops yes, you're right. misread that